front/src/pods/trainer/trainer.container.tsx

```diff
@@ -1,8 +1,84 @@
 import * as React from 'react';
-import {TrainerComponent} from './trainer.component';
+import { useParams } from 'react-router';
+import { baseApiUrl } from 'core/const';
+import { routes } from 'core/router/routes';
+import {
+  createSocket,
+  SocketOuputMessageLiteral,
+  SocketEmitMessageTypes,
+  SocketReceiveMessageTypes,
+} from 'core';
+import { TrainerComponent } from './trainer.component';
+
+interface Params {
+  token: string;
+  room: string;
+}
+
+const useLog = () => {
+  const [log, internalSetLog] = React.useState('');
+  const logRef = React.useRef('');
+
+  const appendToLog = (value: string) => {
+    const newText = `${logRef.current}\n${value} `;
+    internalSetLog(newText);
+    logRef.current = newText;
+  };
+
+  return { log, appendToLog, logRef };
+};
+
 export const TrainerContainer = () => {
+  const { token, room } = useParams<Params>();
+  const { log, appendToLog } = useLog();
+  const [trainerText, setTrainerText] = React.useState<string>('');
+  const [socket, setSocket] = React.useState<SocketIO.Socket>(null);
+  const currentTrainerUrl = `${baseApiUrl}/#${routes.trainer(room, token)}`;
+  const currentStudentUrl = `${baseApiUrl}/#${routes.student(room)}`;
```

Comment:
Include this inside a React.useEffect, []

convert urls current to React.useState

Reply:
```tsx
const [currentTrainerUrl] = React.useState(() => `${baseApiUrl}/#${routes.trainer(room, token)}`);
```